src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -35,95 +33,1003 @@ const ActorInit En_Sw_InitVars = {
+        this->unk_442 = Rand_S16Offset(20, 10);
+        this->unk_444 = func_80B0DE34(this, &this->actor.home);
+        this->unk_448 = this->actor.home.pos;
+        this->actionFunc = func_80B0E9BC;
+        return;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0D14C.s")
+    func_80B0E314(this, this->unk_448, 8.0f);
+    if (DECR(this->unk_440) == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALWALL_DASH);
+        this->unk_440 = 4;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0D364.s")
+    if (!(13.0f < Math_Vec3f_DistXYZ(&this->actor.world.pos, &this->unk_448)) || (func_8002DDF4(globalCtx) != 0)) {
+        this->actionFunc = func_80B0E90C;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0D3AC.s")
+void func_80B0E90C(EnSw* this, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0D590.s")
+    func_80B0E314(this, this->unk_448, 0.0f);
+    if (this->actor.speedXZ == 0.0f) {
+        this->unk_444 = func_80B0DE34(this, &this->actor.home);
+        this->unk_448 = this->actor.home.pos;
+        this->actionFunc = func_80B0E9BC;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0D878.s")
+void func_80B0E9BC(EnSw* this, GlobalContext* globalCtx) {
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0DB00.s")
+    if (func_80B0E430(this, 6.0f, 0x3E8, 0, globalCtx)) {
+        func_80B0E314(this, this->unk_448, 2.0f);
+        if (!(Math_Vec3f_DistXYZ(&this->actor.world.pos, &this->unk_448) > 4.0f)) {
+            this->actionFunc = func_80B0E5E0;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0DC7C.s")
+void EnSw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnSw* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0DE34.s")
+    SkelAnime_Update(&this->skelAnime);
+    func_80B0C9F0(this, globalCtx);
+    this->actionFunc(this, globalCtx);
+    func_80B0CBE8(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0DEA8.s")
+s32 EnSw_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dListP, Vec3f* pos, Vec3s* rot, void* thisx) {
+    Vec3f sp7C = { 1400.0f, -2600.0f, -800.0f };
+    Vec3f sp70 = { 1400.0f, -1600.0f, 0.0f };
+    Vec3f sp64 = { -1400.0f, -2600.0f, -800.0f };
+    Vec3f sp58 = { -1400.0f, -1600.0f, 0.0f };
+    Vec3f sp4C = { 0.0, 0.0f, -600.0f };
+    EnSw* this = THIS;
+    Vec3f sp3C = { 0.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0DFFC.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_sw.c", 2084);
+    if (((this->actor.params & 0xE000) >> 0xD) != 0) {
+        switch (limbIndex) {
+            case 23:
+                *dListP = &D_06004788;
```

Comment:
```suggestion
                *dListP = D_06004788;
```
dLists are already arrays.